Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/BasketViewController.m

```diff
@@ -45,6 +52,20 @@ - (IBAction)donateButtonTapped:(id)sender {
     [self performSegueWithIdentifier:@"BasketPaymentSegue" sender:nil];
 }
 
+- (void)favoriteUnfavoriteBasket {
+    NSMutableArray<Basket*> *favBaskets = [[User currentUser] favoriteBaskets];
+    if ([favBaskets containsObject:self.basket]) {
```

Comment:
There could be consistency issues here too, but only if the same user is logged in on two different devices so I'm not too worried about this case, still wanted to point it out so you're aware.